projects/pastanaga-angular/src/lib/modal/modal.service.ts

```diff
@@ -27,46 +26,49 @@ export class ModalService {
         private zone: NgZone
     ) {}
 
-    openModal(component: Type<IModal>, config?: ModalConfig): ModalRef {
+    openModal(component: Type<any>, config?: ModalConfig): ModalRef {
+        // create the modal reference
+        const ref = new ModalRef({ id: counter++, config });
+        ref.onClose.subscribe(() => this.closeModal(ref));
+
+        // instantiate component
+        const injector = Injector.create({
+            providers: [
+                {
+                    provide: ModalRef,
+                    useValue: ref,
+                },
+            ],
+            parent: this.injector,
+        });
+
         // instantiate modal component
```

Comment:
you should remove this comment have you have it above the injector instantiation

Reply:
Leaving this one, but correcting the one above